server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -233,6 +233,15 @@
         Setting.Property.NodeScope
     );
 
+    /**
+     * Setting to set batch size of stale blobs to be deleted.
+     */
+    public static final Setting<Integer> MAX_STALE_BLOB_DELETE_BATCH_SIZE = Setting.intSetting(
+        "max_stale_blob_delete_batch_size",
+        1000,
```

Comment:
this batch size is used for deleting the unreferenced shard blobs because of snapshot deletion. So probably a better name would be `max_shard_blob_delete_batch_size`. Also is default of 1000 chosen based on some experiment ?

Reply:
Default size of `1000` is chosen because for almost all remote storage, max batch size of bulk delete object is `1000`
References: 
S3 : https://docs.aws.amazon.com/AmazonS3/latest/API/API_DeleteObjects.html
And internally GCP and Azure have 1000 and 100 object delete limit internally.